README.md

```diff
@@ -5,106 +5,143 @@ This is the server component of the AeroGear Metrics Service. It is a RESTful AP
+```
 
-This means that these ports could be in use if you have another postgres instance running or other test web servers.
+### Run Entire Application with Docker Compose
 
-### Running from source
+This section shows how to start the entire application with `docker-compose`. This is useful for doing some quick tests (using the SDKs) for example.
 
-Utilize the `go run` command to transparently compile and run the project:
+First, compile a Linux compatible binary:
 
 ```
-go run cmd/metrics-api/metrics-api.go
+make build_linux
 ```
 
-You can verify the server is running by GET-ing its health check endpoint:
+This binary will be used to build the Docker image. Now start the entire application.
+
 ```
-curl http://localhost:3000/healthz
+docker-comose up
 ```
 
-The default configuration will allow the application to connect to the PostgreSQL container.
+### Example Client Request
+
+This section shows an example `curl` request which can be used to send some data to the `/metrics` endpoint.
 
-### Running a locally-built binary in a container
+```
+curl -i -X POST \
+  http://localhost:3000/metrics \
+  -H 'Cache-Control: no-cache' \
+  -H 'Content-Type: application/json' \
+  -H 'Postman-Token: 87bf2b99-7cdc-8df9-9b2d-6cdcd2932159' \
+  -d '{
+  "clientId": "453de7432",
+  "data": {
+    "app": {
+      "id": "com.example.someApp",
+      "sdkVersion": "2.4.6",
+      "appVersion": "256"
+    },
+    "device": {
+      "platform": "android",
+      "platformVersion": "27"
+    }
+  }
+}'
+```
 
-In order to run a locally-built binary in a container utilize the main `Dockerfile` via :
+You will see the returned back in the response. 
+
+If you have the `psql` command line tools you can connect to the Database and verify the data was inserted.
 
 ```
-make docker_build
+psql -U postgresql -d aerogear_mobile_metrics --host localhost
+Password for user postgresql: # postgres
+
+aerogear_mobile_metrics=> select * from mobileappmetrics;
 ```
 
-This will copy a binary from the default output location for `make build_linux` command, and build an image from it.
+## Builds and Testing
 
-## Release
+The `makefile` provided provides commands for building and testing the code. For now, only the most important commands are documented.
 
-Build and publish to github releases using `goreleaser`, see `.goreleaser.yml` for configuration.
+* `make setup` - Downloads the application dependencies.
 
-First make sure you have this tool installed: [Intalling Goreleaser](https://goreleaser.com/#introduction.installing_goreleaser).
+* `make build` - Compiles a binary for your current system. The binary is output at `./aerogear-app-metrics`.
 
-Then tag your release, replacing `x` with the appropriate version:
-```
-git tag -a v0.0.x -m "Release 0.0.x"
-```
+* `make build_linux` - Compiles a Linux compatible binary. This is mainly used for Docker builds. **Note:** `make build` should still be used if you are a Linux user.
```

Comment:
I think the **Note** here is a bit confusing. Maybe better to just highlight that `build_linux` puts the binary in a different place.